### DISCUSSION TITLE
Reference pressure

### DISCUSSION THREAD

**Discussion by jeromevetel** [xcompact3d/Incompact3d#219] [General]
Hello!

What is the reference pressure in the code? Volume-averaged or something? Usually I take the reference from a point outside any fluctuations, but now I use turbulent data for inlet boundary conditions, then I can not use this anymore.

Thanks!
jerome

**Comment by jeromevetel** [comment]
I would be very grateful if this question could be addressed. I suppose someone has the same interrogation before, right? Thanks!

**Reply by slaizet** [reply]
The pressure is defined up to a constant so you can take any value. If you cannot use a value in an irrotational part of the domain, then you can use a time-averaged value at your inlet.
Best,

Sylvain

**Reply by jeromevetel** [reply]
Sorry for my misunderstanding Sylvain, but the reference changes for each time step, so I don't really understand how a time-averaged value could solve this. I have probably missed something.

jerome

**Reply by slaizet** [reply]
I would assume that the reference pressure would not change during the simulation once the flow is fully established, similar to what you would have in a wind tunnel. 
Best,
Sylvain

**Reply by jeromevetel** [reply]
Many thanks Sylvain. What we observe is that a small fluctuation is always present in our simulations, and as we would like to analyze pressure fluctuations, we need to be very careful in the treatment of it. This is why I asked if the reference pressure could be computed accurately. We will have a closer look on it.

Thanks!
jerome

**Comment by mathrack** [comment]
Hello,

The answer you are seeking is probably located in the subroutine `solve_poisson` located here : https://github.com/xcompact3d/Incompact3d/blob/master/src/navier.f90#L23

Cheers,
Cédric
